File: flureadium/android/src/main/kotlin/dk/nota/flureadium/ControlPanelInfoType.kt
package dk.nota.flureadium

enum class ControlPanelInfoType {
    STANDARD,
    STANDARD_WCH,
    CHAPTER_TITLE_AUTHOR,
    CHAPTER_TITLE,
    TITLE_CHAPTER;

    companion object {
        fun fromString(value: String): ControlPanelInfoType = when (value) {
            "standard" -> STANDARD
            "standardWCh" -> STANDARD_WCH
            "chapterTitleAuthor" -> CHAPTER_TITLE_AUTHOR
            "chapterTitle" -> CHAPTER_TITLE
            "titleChapter" -> TITLE_CHAPTER
            else -> STANDARD
        }

        fun toString(type: ControlPanelInfoType): String = when (type) {
            STANDARD -> "standard"
            STANDARD_WCH -> "standardWCh"
            CHAPTER_TITLE_AUTHOR -> "chapterTitleAuthor"
            CHAPTER_TITLE -> "chapterTitle"
            TITLE_CHAPTER -> "titleChapter"
        }
    }

}

File: flureadium/ios/flureadium/Sources/flureadium/model/ControlPanelInfoType.swift
public enum ControlPanelInfoType {
  case standard
  case standardWCh
  case chapterTitleAuthor
  case chapterTitle
  case titleChapter

  init(from string: String?) {
    switch string {
    case "standard": self = .standard
    case "standardWCh": self = .standardWCh
    case "chapterTitleAuthor": self = .chapterTitleAuthor
    case "chapterTitle": self = .chapterTitle
    case "titleChapter": self = .titleChapter
    default: self = .standard
    }
  }
}
